pkg/machineinformer/relocatablesysfs.go

```diff
@@ -0,0 +1,379 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// derived from https://github.com/google/cadvisor/blob/master/utils/sysfs/sysfs.go @ ef7e64f9
+// as Apache 2.0 license allows.
+
+// relocatablesysfs allows to consume a sysfs tree whose root is not `/sys`.
```

Comment:
Would moving this close to the struct definition allow proper golang docs to be generated (eventually or locally?)

Reply:
we can, but we don't have a godoc reference yet. Docstrings are pretty scattered. Anyway, let me start with this.